### PR TITLE
feat(nat/transitip): add new resource support

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Add new resource support for transit IP of private NAT.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource support.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/nat' TESTARGS='-run=TestAccPrivateTransitIp_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/nat -v -run=TestAccPrivateTransitIp_basic -timeout 360m -parallel 4
=== RUN   TestAccPrivateTransitIp_basic
=== PAUSE TestAccPrivateTransitIp_basic
=== CONT  TestAccPrivateTransitIp_basic
--- PASS: TestAccPrivateTransitIp_basic (64.89s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/nat       64.989s
```

### CODE REVIEW

File: huaweicloud/services/nat/resource_huaweicloud_nat_private_transit_ip.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+				ForceNew:    true,
+				Description: "The ID of the enterprise project to which the transit IP belongs.",
+			},
+			"tags": common.TagsSchema(),
+			"network_interface_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The network interface ID of the transit IP for private NAT.",
+			},
+			"gateway_id": {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/nat/resource_huaweicloud_nat_private_transit_ip.go:63]: missing in doc

File: huaweicloud/services/nat/resource_huaweicloud_nat_private_transit_ip.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	natClient, err := cfg.NatV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating NAT v3 client: %s", err)
+	}
+
+	resp, err := transitips.Get(natClient, d.Id())
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "Transit IP (Private NAT)")
+	}
+	mErr := multierror.Append(nil,
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/nat/resource_huaweicloud_nat_private_transit_ip.go:118]: gateway_id not set

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/nat' TESTARGS='-run=TestAccPrivateTransitIp_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/nat -v -run=TestAccPrivateTransitIp_basic -timeout 360m -parallel 4
=== RUN   TestAccPrivateTransitIp_basic
=== PAUSE TestAccPrivateTransitIp_basic
=== CONT  TestAccPrivateTransitIp_basic
--- PASS: TestAccPrivateTransitIp_basic (63.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/nat       63.415s
```

File: huaweicloud/services/nat/resource_huaweicloud_nat_private_transit_ip.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/nat/resource_huaweicloud_nat_private_transit_ip.go:118]: done.

File: huaweicloud/services/nat/resource_huaweicloud_nat_private_transit_ip.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/nat/resource_huaweicloud_nat_private_transit_ip.go:63]: done.

**Pull request merged**